src/lua/LuaFileSystem.cpp

```diff
@@ -146,9 +208,58 @@ static int l_filesystem_join_path(lua_State *l)
 		lua_pushlstring(l, path.c_str(), path.size());
 		return 1;
 	} catch (const std::invalid_argument &) {
-		luaL_error(l, "result is not a valid path");
-		return 0;
+		return luaL_error(l, "result is not a valid path");
+	}
+}
+
+
+/*
+ * Function: MakeDirectory
+ *
+ * > local success = FileSystem.MakeDirectory( dir_name )
+ *
+ * Creating the given directory if it's missing, returning a boolean
+ * indicating success
+ *
+ * Availability:
+ *
+ *   Oct 2023
+ *
+ * Status:
+ *
+ *   experimental
+ */
+static int l_filesystem_make_directory(lua_State *l)
+{
+	SplitURI split_uri = ParseURIArg(l, 1);
+	const FileSource fs = DetermineFileSource(split_uri.scheme);
+	if (nullptr == fs.source)
+	{
+		return luaL_error(l, "Path '%s' does not contain a valid scheme, must be user:// or data://", split_uri.full_path.data());
+	}
+
+	if (false == fs.is_read_write)
+	{
+		return luaL_error(l, "'%s' does not support file operations that modify it, only things in the user folder do", split_uri.full_path.data());
+	}
```

Comment:
We generally don't use the pattern of `constant == variable` in Pioneer - if there's a concern about the value being overwritten, you could prefer e.g. `!fs.source` or `!fs.is_read_write`. The use of this pattern isn't something that needs to be addressed to merge, just pointing out the de-facto style the rest of the codebase follows.

Reply:
OK, noted, I'll try and keep to the style.

Will take a while to override my default of never using `!` and always using Yoda conditionals I expect :-)